src/projection_client.rs

```diff
@@ -0,0 +1,629 @@
+
+                let _ = client.reset(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn disable_projection<Name>(
+        &self,
+        name: Name,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        self.disable_projection_internal(name, true, credentials)
+            .await
+    }
+
+    pub async fn abort_projection<Name>(
+        &self,
+        name: Name,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        self.disable_projection_internal(name, false, credentials)
+            .await
+    }
+
+    async fn disable_projection_internal<Name>(
+        &self,
+        name: Name,
+        write_checkpoint: bool,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        let options = projections::disable_req::Options {
+            name: name.as_ref().to_string(),
+            write_checkpoint,
+        };
+
+        let req = projections::DisableReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel.clone());
+
+                let _ = client.disable(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn projection_state<Name, Partition, A>(
+        &self,
+        name: Name,
+        partition: Partition,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<serde_json::Result<A>>
+    where
+        Name: AsRef<str>,
+        Partition: AsRef<str>,
+        A: DeserializeOwned + Send,
+    {
+        let options = projections::state_req::Options {
+            name: name.as_ref().to_string(),
+            partition: partition.as_ref().to_string(),
+        };
+
+        let req = projections::StateReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel.clone());
+
+                let resp = client.state(req).await?.into_inner();
+                let value = resp
+                    .state
+                    .map(parse_value)
+                    .unwrap_or(serde_json::Value::Null);
+
+                Ok(serde_json::from_value(value))
+            })
+            .await
+    }
+
+    pub async fn projection_result<Name, Partition, A>(
```

Comment:
**COMMENT:** In .NET this method is called `GetResultAsync`.

Reply:
Renamed `get_result`.